kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/MountedResourceConfigProvider.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.operator;
+
+import java.nio.file.Path;
+import java.util.function.BiFunction;
+
+import io.fabric8.kubernetes.api.model.Volume;
+import io.fabric8.kubernetes.api.model.VolumeBuilder;
+import io.fabric8.kubernetes.api.model.VolumeMount;
+import io.fabric8.kubernetes.api.model.VolumeMountBuilder;
+
+/**
+ * A {@link SecureConfigProvider} for Kubernetes Secrets and ConfigMaps
+ */
+public class MountedResourceConfigProvider implements SecureConfigProvider {
+
+    static final MountedResourceConfigProvider SECRET_PROVIDER = new MountedResourceConfigProvider("", "secrets",
```

Comment:
I think secrets and configmaps are the right place to start   I think what you are doing naturally extends to SAs,  and ClusterTrustBundles. https://kubernetes.io/docs/concepts/storage/projected-volumes/.  Those would be RFEs of course.